packages/headless/src/app/search-engine/jwt-reducer.ts

```diff
@@ -109,10 +116,16 @@ const handleMismatchOnSearchHub = (
     payload,
     logger
   );
-  if (shouldReconcileValues(jwt.searchHub, state.searchHub)) {
-    state.searchHub = jwt.searchHub!;
-  }
+  return updateSearchHub(jwt, state);
```

Comment:
I didn't remove the logic that updated the pipeline/etc from the actions responsible for updating them directly, since their own reducers aren't aware that the access token contains the "correct" value, and would update them.